File: app/src/main/java/com/jidesh/erosnowjidesh/ViewHolders/FavFragViewmodel.java
package com.jidesh.erosnowjidesh.ViewHolders;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.jidesh.erosnowjidesh.Models.Entities.Fav_movies;
import com.jidesh.erosnowjidesh.repository.Fav_repository;

import java.util.List;

public class FavFragViewmodel extends AndroidViewModel {
    private Fav_repository repository;
    private LiveData<List<Fav_movies>> allFavMovies;
    public FavFragViewmodel(@NonNull Application application) {
        super(application);
        repository=new Fav_repository(application);
        allFavMovies=repository.getAllFavs();
    }
    public void inset(Fav_movies fav_movies)
    {
        repository.Insert(fav_movies);
    }
    public void delete(Fav_movies fav_movies)
    {
        repository.Delete(fav_movies);
    }
    public LiveData<List<Fav_movies>> getAllFavMovies()
    {
        return allFavMovies;
    }
    public Integer check(int a)
    {
        return   repository.find(a);

    }
}

File: app/src/main/java/com/jidesh/erosnowjidesh/ViewHolders/FavFragViewmodel.kt
package com.jidesh.erosnowjidesh.ViewHolders

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.jidesh.erosnowjidesh.Models.Entities.Fav_movies
import com.jidesh.erosnowjidesh.repository.Fav_repository

class FavFragViewmodel(application: Application) : AndroidViewModel(application) {
    private val repository: Fav_repository
    val allFavMovies: LiveData<List<Fav_movies>>?
    fun inset(fav_movies: Fav_movies?) {
        repository.Insert(fav_movies)
    }

    fun delete(fav_movies: Fav_movies?) {
        repository.Delete(fav_movies)
    }

    fun check(a: Int): Int {
        return repository.find(a)
    }

    init {
        repository = Fav_repository(application)
        allFavMovies = repository.allFavs
    }
}